src/animation/fluidResistance.ts

```diff
@@ -0,0 +1,139 @@
+import { rAF } from '../rAF';
+import {
+  Entity,
+  applyForce,
+  fluidResistanceForceV,
+  gE,
+  getFluidPositionAtTerminalVelocity,
+} from '../forces';
+import { vector as Vector, addf, multf } from '../core';
+import { Listener, AnimationParams, AnimationInitializer } from './types';
+
+export interface FluidResistance1DParams extends AnimationParams {
+  config: {
+    mass: number;
+    rho: number;
+    area: number;
+    cDrag: number;
+    settle?: boolean;
+  };
+}
+
+interface FluidResistanceState {
+  mover: Entity;
+  startTime: DOMHighResTimeStamp;
+}
+
+/**
+ * A function to apply the drag force on each step in the course
+ * of the animation. First, we derive the force vector exerted by the
+ * fluid on the mover. Then we apply that vector to the mover to determine
+ * its next acceleration, velocity, and position.
+ */
+const applyFluidResistanceForceForStep = (
+  { mover }: FluidResistanceState,
+  config: FluidResistance1DParams['config']
+) => {
+  const dragForce = fluidResistanceForceV({
+    rho: config.rho,
+    area: config.area,
+    velocity: mover.velocity,
+    cDrag: config.cDrag,
+  });
+
+  const gravitationalForce: Vector<number> = [0, mover.mass * gE];
+  const force = multf({
+    v: addf({ v1: dragForce, v2: gravitationalForce }),
+    s: 0.001,
```

Comment:
This is something I've been missing in other equations that I think we'll need to backport. Basically, our force vector is calculated using seconds, but is being _applied_ roughly once every ms. Therefore, we need to scale the vector down (multiply by 0.001) to achieve the force that would be applied every ms.

Reply:
Good catch with this bug, I think you're correct, all the previous `applyForce` calls are being called every millisecond but applying 1 full second worth of acceleration & velocity :scream:

However, I don't think scaling down the force is applying the math correctly throughout the `applyForce` function.  I'll try and explain why I think so below:

```
Below are what I believe are the methods to calculate acceleration, velocity, and distance:

acc = F/m
vel = currVel + (acc * t)
pos = currPos + (vel * t)

In the current implementation of applyForce, you are _not_ using t.  Which implies that t = 1.  

So the formulas currently being used are:

acc = F/m
vel = currVel + acc ==> currVel + (acc * 1)
pos = currPos + vel ===> currPos + (vel * 1)

Your fix introduces a t = 0.001 multiplication at the F (force) level, but I don't think it propagates down correctly to the position.  

Here's my logic, by multiplying F by a t of 0.001, the applyForce formula updates as follows:

accWithT = (F * t)/m ==> (F/m) * t ==> acc * t (which is just velocity)
velWithT =  currVel + accWithT ==> currVel + (acc * t)
pos = currPos + velWithT ===> currPos + (velWithT * 1) [😱a t = 1 is still being applied to position calculation] 
```
By multiplying the original F with t, you're changing the acceleration calculation in applyForce to be a velocity calculation, this fixes the nextVelocity calculation (because its now adding a `acc * t` instead of `acc * 1`), but it doesn't address the final position calculation which is still using `velocity * 1`.

All this to say that, I think you found a legit bug in the calculations, awesome find!  But I think the solution to fix all the things would be to update `applyForce` to accept `force`, `entity`, and `duration`.  And then update the `nextVelocity` and `nextPosition` calculations to use the passed in duration (i.e. v = v0 + at, p = p0 + vt).

And then all you'd have to do is pass in duration = 0.001 to any call to `applyForce` and boom, that should fix the issue of calling each step on a millisecond basis but applying the acceleration & velocity at per second basis.

Reply:
Thanks so much for typing up your thought process @mariano-formidable ❤️ I think you hit the nail on the head, and this would explain some of the challenges I've run into moving this same setup over to friction. I'm going to make the change just for fluid resistance in this PR, then try to get it ported over for the other two in the next PR (which will also introduce the `infinite` flag).